capsule-server-import-export.php

```diff
@@ -1,211 +1,78 @@
-									$terms_as_id[] = $term_id;
-								}
-							}
-							
-							wp_set_post_terms($this->local_post_id, $terms_as_id, $taxonomy);
-						}
-						else {
-							wp_set_post_terms($this->local_post_id, $terms, $taxonomy);
-						}
-					}
-				}
-				else {
-					// There were no terms passed, so unset the current terms
-					wp_set_post_terms($this->local_post_id, array(), $taxonomy);
-				}
-			}
-		}
-	}
-
-	public static function get_post_id_by_guid($guid) {
-		global $wpdb;
-		$sql = $wpdb->prepare("
-			SELECT ID
-			FROM $wpdb->posts
-			WHERE guid = %s
-		", $guid);
-
-		$post_id = $wpdb->get_var($sql);
-
-		if (!empty($post_id)) {
-			return $post_id;
-		}
-		else {
-			return false;
-		}
-	}
-
-	//@TODO remove user from blog, delete their API key
-	function cap_server_remove_user() {}
-}
-
-class Capsule_Server_Export_Terms {
-	
-	function __construct($taxonomies = array()) {
-		$this->taxonomies = $taxonomies;
-	}
-
-	/**
-	 * @return array of formatted taxonomies:
-	 *  'taxonomy_1' => array(
-	 *		'term-slug' => array(
-	 *			'id' => 1,
-	 *			'name' => 'Amazing Term',
-	 *			'description' => 'This term is amazing AND useful',
-	 *		),
-	 *		'term-slug-2' ...
-	 * 	),
-	 * 	'taxonomy_2' ...
-	 */ 
-	function get_terms() {
-		$taxonomy_array = array();
-
-		$terms = get_terms($this->taxonomies, array(
-			'hide_empty' => false,
-			'orderby' => 'slug',
-			'order' => 'ASC',
-		));
-		if (is_array($terms)) {
-			foreach ($terms as $term) {
-				$taxonomy_array[$term->taxonomy][$term->slug] = array(
-					'id' => $term->term_id, 
-					'name' => $term->name,
-					'description' => $term->description,
-					'taxonomy' => $term->taxonomy,
-				);
-			}
-		}
-		return $taxonomy_array;
-	}
-}
-
+<?php
+/**
+ * Implement post import and taxonomy terms export.
+ */
+
+require_once 'inc/class-capsule-server-import-post.php';
+require_once 'inc/class-capsule-server-export-terms.php';
+
+/**
+ * Import export controller function.
+ *
+ * @return void
+ */
 function capsule_server_controller() {
 	if ( empty( $_POST['capsule_server_action'] ) ) {
 		return;
 	}
 
-	switch ($_POST['capsule_server_action']) {
+	switch ( $_POST['capsule_server_action'] ) {
 		case 'insert_post':
 			$response = array(
 				'result' => 'error',
 			);
 			// Cannot use nonce here as they're salted with unique keys, going to have to rely on api key.
-			if (isset($_POST['capsule_client_post_data'])) {
+			if ( isset( $_POST['capsule_client_post_data'] ) ) {
 				$data = $_POST['capsule_client_post_data'];
-				if (isset($data['post']) && isset($data['tax']) && isset($data['api_key'])) {
-					if ($user_id = capsule_server_validate_user($data['api_key'])) {
-						$capsule_import = new Capsule_Server_Import_Post($user_id, $data['post'], $data['tax']);
-						$post_id = $capsule_import->import();
-						
-						if ($capsule_import->local_post_id != 0) {
+				if ( isset( $data['post'] ) && isset( $data['tax'] ) && isset( $data['api_key'] ) ) {
```

Comment:
These could be put into a single `isset` call like `isset( $data['post'], $data['tax'], $data['api_key'] )`.